File: files/CH-132.py
import pandas as pd
import re

path = "CH-132 Merge.xlsx"

input = pd.read_excel(path, usecols="B:C", skiprows=1, nrows=5)
input2 = pd.read_excel(path, usecols="H:H", skiprows=1, nrows=7)
test = pd.read_excel(path, usecols="I:I", skiprows=1, nrows=7).rename(columns=lambda x: x.replace('.1', '')).sort_values(by='Value', ascending=False).reset_index(drop=True)

result = input.merge(input2, how='cross')
result['Match'] = result.apply(lambda x: bool(re.search(x['Sub code'], x['Code'])), axis=1)
result = result[result['Match']].sort_values(by='Value', ascending=False).reset_index(drop=True)[['Value']]
 
print(result.equals(test)) # True

File: files/100-199/132/CH-132.R
library(tidyverse)
library(readxl)

path = "files/CH-132 Merge.xlsx"
input = read_excel(path, range = "B2:C7")
input2 = read_excel(path, range = "H2:H8")
test  = read_excel(path, range = "I2:I8") %>% arrange(desc(Value))

result = input %>%
  cross_join(input2) %>%
  filter(str_detect(Code, `Sub code`)) %>%
  select(Value) %>%
  arrange(desc(Value))

all.equal(result, test, check.attributes = FALSE)
#> [1] TRUE